File: scratch/find_rpcs.py
import os
import re

def search_rpc(directory):
    pattern = re.compile(r'\.rpc\s*\(\s*[\'"`](.*?)[\'"`]')
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith('.js') or file.endswith('.jsx'):
                path = os.path.join(root, file)
                try:
                    with open(path, 'r', encoding='utf-8') as f:
                        content = f.read()
                        matches = pattern.findall(content)
                        for m in matches:
                            print(f"Found RPC call '{m}' in {path}")
                except Exception as e:
                    pass

search_rpc(r'C:\Users\edarj\.gemini\antigravity\scratch\grade-management-system\src')

File: scratch/find_rpcs.js
const fs = require('fs');
const path = require('path');

function searchRPC(dir) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat.isDirectory()) {
      searchRPC(fullPath);
    } else if (stat.isFile() && (file.endsWith('.js') || file.endsWith('.jsx'))) {
      const content = fs.readFileSync(fullPath, 'utf8');
      const regex = /\.rpc\s*\(\s*['"`](.*?)['"`]/g;
      let match;
      while ((match = regex.exec(content)) !== null) {
        console.log(`Found RPC call "${match[1]}" in ${fullPath}`);
      }
    }
  }
}

searchRPC('C:\\Users\\edarj\\.gemini\\antigravity\\scratch\\grade-management-system\\src');
